tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/LanguageRepoService.cs

```diff
@@ -0,0 +1,121 @@
+{
+    protected readonly string _packagePath;
+
+    public LanguageRepoService(string packagePath)
+    {
+        _packagePath = packagePath ?? throw new ArgumentNullException(nameof(packagePath));
+    }
+
+    /// <summary>
+    /// Creates a success response.
+    /// </summary>
+    /// <param name="message">Success message</param>
+    /// <param name="exitCode">Exit code (default: 0 for success)</param>
+    /// <returns>SuccessCLICheckResponse with success status and response</returns>
+    protected static SuccessCLICheckResponse CreateSuccessResponse(string message, int exitCode = 0)
+    {
+        return new SuccessCLICheckResponse(exitCode, message);
+    }
+
+    /// <summary>
+    /// Creates a failure response.
+    /// </summary>
+    /// <param name="message">Failure message</param>
+    /// <param name="exitCode">Exit code (default: 1 for failure)</param>
+    /// <param name="error">Additional error details</param>
+    /// <returns>FailureCLICheckResponse with failure status and response</returns>
+    protected static FailureCLICheckResponse CreateFailureResponse(string message, int exitCode = 1, string error = "")
+    {
+        return new FailureCLICheckResponse(exitCode, message, error);
+    }
+
+    /// <summary>
+    /// Creates a cookbook response.
+    /// </summary>
+    /// <param name="cookbookReference">Reference to cookbook or documentation</param>
+    /// <param name="message">Additional response message</param>
+    /// <param name="exitCode">Exit code (default: 0 for success)</param>
+    /// <returns>CookbookCLICheckResponse with cookbook reference and response</returns>
+    protected static CookbookCLICheckResponse CreateCookbookResponse(string cookbookReference, string message, int exitCode = 0)
+    {
+        return new CookbookCLICheckResponse(exitCode, message, cookbookReference);
+    }
+
+    public virtual async Task<ICLICheckResponse> AnalyzeDependenciesAsync(CancellationToken ct)
+    {
+        await Task.CompletedTask;
+        return CreateFailureResponse("AnalyzeDependencies not implemented for this language");
+    }
+
+    public virtual async Task<ICLICheckResponse> FormatCodeAsync()
+    {
+        await Task.CompletedTask;
+        return CreateFailureResponse("FormatCode not implemented for this language");
+    }
+
+    public virtual async Task<ICLICheckResponse> LintCodeAsync()
+    {
+        await Task.CompletedTask;
+        return CreateFailureResponse("LintCode not implemented for this language");
+    }
+
+    public virtual async Task<ICLICheckResponse> RunTestsAsync()
+    {
+        await Task.CompletedTask;
+        return CreateFailureResponse("RunTests not implemented for this language");
+    }
+
+    protected ICLICheckResponse CreateResponse(string prefix, int exitCode, string output)
+    {
+        if (exitCode == 0)
+        {
+            return CreateSuccessResponse($"{prefix} completed successfully.{Environment.NewLine}{output}");
```

Comment:
Same as with the failure response, I think we can just remove these. I'd prefer to compose stuff directly on the properties instead of building strings at different layers. We should have two stages with the response classes: property setting and then final formatting.